databricks_cli/stack/api.py

```diff
@@ -0,0 +1,348 @@
+        the deployment will abort.
+
+        :param job_settings:
+        :return: job_id, Physical ID of job on Databricks server.
+        """
+        if 'name' not in job_settings:
+            raise StackError("Please supply 'name' in job resource 'resource_properties'")
+        jobs_same_name = self.jobs_client._list_jobs_by_name(job_settings['name'])
+        if len(jobs_same_name) > 1:
+            raise StackError('Multiple jobs with the same name already exist, aborting job'
+                             ' resource deployment')
+        elif len(jobs_same_name) == 1:
+            first_job = jobs_same_name[0]
+            creator_name = first_job['creator_user_name']
+            timestamp = first_job['created_time'] / MS_SEC  # Convert to readable date.
+            date_created = datetime.fromtimestamp(timestamp).strftime('%Y-%m-%d %H:%M:%S')
+            click.echo('Warning: Job exists with same name created by %s on %s. Job will '
+                       'be overwritten' % (creator_name, date_created))
+            self.update_job(job_settings, first_job['job_id'])
+            return first_job['job_id']
+        else:
+            click.echo("Creating new job")
+            job_id = self.jobs_client.create_job(job_settings)['job_id']
+            return job_id
+
+    def update_job(self, job_settings, job_id):
+        """
+        Given job settings and an existing job_id of a job, update the job settings on databricks.
+
+        :param job_settings: job settings to update the job with.
+        :param job_id: physical job_id of job in databricks server.
+        """
+        click.echo("Updating Job")
+        self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings})
+
+    def deploy_job(self, resource_id, resource_properties, physical_id=None):
+        """
+        Deploys a job resource by either creating a job if the job isn't kept track of through
+        the physical_id of the job or updating an existing job. The job is created or updated using
+        the the settings specified in the inputted job_settings.
+
+        :param resource_id: The stack-internal resource ID of the job.
+        :param resource_properties: A dict of the Databricks JobSettings data structure
+        :param physical_id: A dict object containing 'job_id' field of job identifier in Databricks
+        server
+
+        :return: tuple of (physical_id, deploy_output), where physical_id contains a 'job_id' field
+        of the physical job_id of the job on databricks. deploy_output is the output of the job
+        from databricks when a GET request is called for it.
+        """
+        job_settings = resource_properties  # resource_properties of jobs are solely job settings.
+        click.echo("Deploying job '%s' with settings: \n%s \n" % (resource_id, json.dumps(
+            job_settings, indent=2, separators=(',', ': '))), nl=False)
+
+        if physical_id and 'job_id' in physical_id:
```

Comment:
I still wouldn't want to throw if job_id is not found, because the creation of the stack status is beyond the user's control and I wouldn't want to throw if there's a problem with a file that the user didn't generate themselves

Reply:
I'll need to think about this more actually. it's probably best to throw an error